ide/o.n.swing.dirchooser/src/org/netbeans/swing/dirchooser/DirectoryChooserUI.java

```diff
@@ -296,9 +299,16 @@ private void updateUseShellFolder() {
         }
     }
 
-    /** Returns instance of WindowsPlacesBar class or null in case of failure
+    /** 
+     * Returns a shortcuts panel, e.g the windows places bar or the NB favorites shortcuts panel
      */
-    private JComponent getPlacesBar () {
+    private JComponent getPlacesBar() {
+        if (useNBShortcutsPanel()) {
+            FlatShortcutsPanel shortcuts = new FlatShortcutsPanel(fileChooser);
+            JScrollPane sp = new JScrollPane(shortcuts, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+            sp.setPreferredSize(new Dimension(100, 300));
+            return sp;
+        }
```

Comment:
@DevCharly do you think it would make sense for FlatLaf to wrap the `FlatShortcutsPanel` in a JScrollPane like here?

NB uses two file choosers, one is customized and used for opening projects etc, the other one is the standard JDK file chooser. Now when the shortcuts panel has lots of items the standard JDK file chooser will get very tall.

Reply:
filed https://github.com/JFormDesigner/FlatLaf/issues/828